pkg/volume/iscsi/disk_manager.go

```diff
@@ -96,7 +97,9 @@ func diskSetUp(manager diskManager, b iscsiDiskMounter, volPath string, mounter
 	}
 
 	if !b.readOnly {
-		volume.SetVolumeOwnership(&b, volPath, fsGroup, fsGroupChangePolicy, util.FSGroupCompleteHook(b.plugin, nil))
+		// This code requires larger refactor to monitor progress of ownership change
+		ownershipChanger := volume.NewVolumeOwnership(&b, volPath, fsGroup, fsGroupChangePolicy, util.FSGroupCompleteHook(b.plugin, nil))
+		_ = ownershipChanger.ChangePermissions()
```

Comment:
Note to reviewers: Historically `fc`, `iscsi` and `emptydir` ignored returned errors from `SetVolumeOwnership` function. I think that is an oversight. CSI plugin, marks such volumes as mounted in uncertain state. I will file follow up issues to fix this behaviour. I didn't wanted to overly complicate this PR.

Reply:
FWIW, in your future PR, I don't think we should do any more than report any errors.

If there's actual disk corruption (or some NFS weirdness), then the the volume wouldn't have mounted.

Otherwise an error with chmod is probably obscure & benign, and we would still want the volume to mount since in most cases I would think the workload would run.

Usually the game is to try and get volumes attached and workloads running. We don't need more reasons IMHO for workloads to not run.

Reply:
I am not so sure about that. If somehow `chmod` and `chown` failed on the volume and then workload fails unpredictably while running, isn't that worse? I would rather fail hard than fail unpredictably if that makes sense.

Reply:
It depends on how exactly they fail. I think the on-root-mismatch flag is a strong signal that users care more about the volume getting mounted than the chmod being exactly correct everywhere.

Anyway, let's continue this discussion, it's not blocking this PR.